packages/lesswrong/server/posts/annualReviewMarkets.ts

```diff
@@ -0,0 +1,70 @@
+import { AnnualReviewMarketInfo } from "../../lib/annualReviewMarkets";
+import ManifoldProbabilitiesCaches from "../../lib/collections/manifoldProbabilitiesCaches/collection";
+import { createAdminContext, createMutator } from "../vulcan-lib";
+
+const postGetMarketInfoFromManifold = async (post: DbPost): Promise<AnnualReviewMarketInfo | null > => {
+  if (!post.manifoldReviewMarketId) return null;
+
+  const result = await fetch(`https://api.manifold.markets./v0/market/${post.manifoldReviewMarketId}`, {
+    method: "GET",
+    headers: {
+      "content-type": "application/json"
+    },
+  })
+  
+  if (!result.ok) throw new Error(`HTTP error! status: ${result.status}`);
+
+  const fullMarket = await result.json()
+  if (fullMarket.probability == null) throw new Error("Manifold market probability is null");
+  if (fullMarket.isResolved == null) throw new Error("Manifold market isResolved is null");
+
+  return { probability: fullMarket.probability, isResolved: fullMarket.isResolved, year: post.postedAt.getFullYear() }
+}
+
+
+async function updateMarketInfoInCache(post: DbPost) {
+  const marketInfo = await postGetMarketInfoFromManifold(post);
+  if (!marketInfo || !post.manifoldReviewMarketId) return null;
+
+  await createMutator({
+    collection: ManifoldProbabilitiesCaches,
+    document: {
+      marketId: post.manifoldReviewMarketId,
+      probability: marketInfo.probability,
+      isResolved: marketInfo.isResolved,
+      year: marketInfo.year,
+      lastUpdated: new Date(),
+    },
+    context: createAdminContext(),
+  }).catch(error => {
+    // eslint-disable-next-line no-console
+    console.error("Failed to update cache for post:", post._id, error.data.errors);
+  })
```

Comment:
I think if we want to swallow failures, we probably don't want to do it at this level?